BookingSystem/BookingSystem.Commands/Infrastructure/Result.cs

```diff
@@ -0,0 +1,44 @@
+﻿using BookingSystem.Commands.Properties;
+using BookingSystem.WritePersistence.WriteModels;
+
+namespace BookingSystem.Commands.Infrastructure
+{
+
+    public class Result<T>
+    {
+        public T Value { get; set; }  
+        public bool IsSuccessful { get; set; }
+        public string ErrorMessage { get; set; } 
+        //Todo: List<string> Errors{ get; set; }
+        public Result() { }
+        public Result(T value, bool isSuccessful, string errorMessage)
+        {
+            Value = value;
+            IsSuccessful = isSuccessful;
+            ErrorMessage = errorMessage;
+        }
+
+        public static Result<T> Ok(T value) => new Result<T>(value, true, null);
+       
+    }
+
+    public class Result : Result<string>
```

Comment:
how are going to use this result? I understand that as Result doesn't have something custom to return except the status and the error. Generic Result<T> extends Result  with returned value. Does it look different for you?